sync-core/src/main/java/com/cloudant/sync/datastore/DatastoreManager.java

```diff
@@ -153,8 +156,8 @@ public String getPath() {
     public Datastore openDatastore(String dbName) throws DatastoreNotCreatedException {
         Preconditions.checkArgument(dbName.matches(LEGAL_CHARACTERS),
                 "A database must be named with all lowercase letters (a-z), digits (0-9),"
-                  + " or any of the _$()+-/ characters. The name has to start with a"
-                  + " lowercase letter (a-z).");
+                        + " or any of the _$()+-/ characters. The name has to start with a"
```

Comment:
The one-arg version of `openDatastore` should call the two-arg `openDatastore` with NullKeyProvider as discussed.